### PR TITLE
Increase help content available in Matomo Tag Manager

### PR DESCRIPTION
**Pull request by snake14**
### Description:

Increase help content available in Matomo Tag Manager by adding and updating help and links in the application.
Jira Issue: PG-1320

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: lang/en.json [hunk 1]
```diff
@@ -151,6 +151,11 @@
         "CustomVariables": "Custom variables",
         "CustomizeTracking": "Customize Tracking",
         "CustomizeTrackingTeaser": "Want to configure the Matomo tracking for this container? Here is a list of existing Matomo configurations in this container:",
+        "DashboardHelp": "<p><strong>How does the \"Tag Manager > Dashboard\" page help me?</strong></p><p>The Tag Manager Dashboard provides an instant overview of both the names and quantity of tags, triggers, variables and versions of container configurations that have been created in your Matomo Tag Manager.</p><p>Within the Tag Manager Dashboard, you also can edit or create new tags, triggers, variables and versions.</p><p>Learn more about this in the %1$sTag Manager Dashboard Guide%2$s.</p>",
```
Comment (AltamashShaikh) [inline lang/en.json]: @snake14 We do not include HTML here you can use something like this.
```suggestion
        "DashboardHelp": "%1$s%2$sHow does the \"Tag Manager > Dashboard\" page help me?%3$s%4$s%1$sThe Tag Manager Dashboard provides an instant overview of both the names and quantity of tags, triggers, variables and versions of container configurations that have been created in your Matomo Tag Manager.%4$s%1$sWithin the Tag Manager Dashboard, you also can edit or create new tags, triggers, variables and versions.%4$s%1$sLearn more about this in the %5$sTag Manager Dashboard Guide%6$s.%1$s",
```

Comment (justinvelluppillai) [reply lang/en.json]: Or consider creating one entry per paragraph to separate them. It would lead to better results with the translations also 👍🏽

Comment (AltamashShaikh) [reply lang/en.json]: Yes makes more sense `DashboardHelpPara1`, `DashboardHelpPara2` and so on

Comment (snake14) [reply lang/en.json]: Yeah. I used placeholders for the anchors, but the [developer documentation](https://developer.matomo.org/guides/translations#best-practices-for-new-translation-keys) said that it's alright to use basic tags, so I didn't bother splitting them out. I can break up the translations, if you think I should.

Comment (AltamashShaikh) [reply lang/en.json]: @justinvelluppillai We generally don't include HTML tags right ?
Or this one is okay to pass ?

Comment (justinvelluppillai) [reply lang/en.json]: We do use some, sometimes. In this case I think just creating three translation strings is worthwhile, and serves the purpose of making translations easier to maintain and removing a few tags.

Comment (justinvelluppillai) [reply lang/en.json]: `DashboardHelp1`, `DashboardHelp2`, `DashboardHelp3` is quite common

Comment (AltamashShaikh) [reply lang/en.json]: @snake14 Just split the longer ones into sub paras

Comment (snake14) [reply lang/en.json]: Sounds good. I'll try to do that Monday morning. I'm trying to see if I can get the rest of the issue in another PR before I sign off for the day.

Comment (justinvelluppillai) [reply lang/en.json]: Thanks @snake14, sorry for the inconvenience we can explain/discuss why next week!

**Comment by snake14** [comment]
@AltamashShaikh @justinvelluppillai It took me a little while to figure out a good way to split up those translations since the Vue build process didn't like concatenated strings, but I think I got it.

**Comment by justinvelluppillai** [comment]
Nice one, those strings look like they'll translate much better now 👍🏽

File: Controller.php [hunk 2]
```diff
@@ -93,25 +167,90 @@ public function gettingStarted()
...
+        $tagsHelpText = $this->renderTemplate('helpContent', [
+            'subHeading' => Piwik::translate('TagManager_ManageTagsHelp1'),
+            'paragraphs' => [
+                Piwik::translate('TagManager_ManageTagsHelp2'),
+                Piwik::translate('TagManager_ManageTagsHelp3', [
+                    '<a href="https://matomo.org/guide/tag-manager/tags/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/guide/tag-manager/getting-started-with-tag-manager/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/tag-manager-training/" rel="noreferrer noopener" target="_blank">',
```
Comment (AltamashShaikh) [inline Controller.php:179]: Should we link this to - https://matomo.org/tag-manager-training/#mtm-tags to make it more specific ?

File: Controller.php [hunk 3]
```diff
@@ -93,25 +167,90 @@ public function gettingStarted()
...
+        $triggersHelpText = $this->renderTemplate('helpContent', [
+            'subHeading' => Piwik::translate('TagManager_ManageTriggersHelp1'),
+            'paragraphs' => [
+                Piwik::translate('TagManager_ManageTriggersHelp2'),
+                Piwik::translate('TagManager_ManageTriggersHelp3', [
+                    '<a href="https://matomo.org/guide/tag-manager/triggers/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/guide/tag-manager/getting-started-with-tag-manager/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/tag-manager-training/" rel="noreferrer noopener" target="_blank">',
```
Comment (AltamashShaikh) [inline Controller.php:199]: Similarly - https://matomo.org/tag-manager-training/#mtm-triggers

File: Controller.php [hunk 4]
```diff
@@ -93,25 +167,90 @@ public function gettingStarted()
...
+        $variablesHelpText = $this->renderTemplate('helpContent', [
+            'subHeading' => Piwik::translate('TagManager_ManageVariablesHelp1'),
+            'paragraphs' => [
+                Piwik::translate('TagManager_ManageVariablesHelp2'),
+                Piwik::translate('TagManager_ManageVariablesHelp3', [
+                    '<a href="https://matomo.org/guide/tag-manager/variables/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/guide/tag-manager/getting-started-with-tag-manager/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/tag-manager-training/" rel="noreferrer noopener" target="_blank">',
```
Comment (AltamashShaikh) [inline Controller.php:221]: Similarly - https://matomo.org/tag-manager-training/#mtm-variables

File: Controller.php [hunk 5]
```diff
@@ -93,25 +167,90 @@ public function gettingStarted()
...
+        $versionsHelpText = $this->renderTemplate('helpContent', [
+            'subHeading' => Piwik::translate('TagManager_ManageVersionsHelp1'),
+            'paragraphs' => [
+                Piwik::translate('TagManager_ManageVersionsHelp2'),
+                Piwik::translate('TagManager_ManageVersionsHelp3', [
+                    '<a href="https://matomo.org/guide/tag-manager/versions-and-publish/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/guide/tag-manager/getting-started-with-tag-manager/" rel="noreferrer noopener" target="_blank">',
+                    '</a>',
+                    '<a href="https://matomo.org/tag-manager-training/" rel="noreferrer noopener" target="_blank">',
```
Comment (AltamashShaikh) [inline Controller.php:245]: https://matomo.org/tag-manager-training/#mtm-versions

**Review by AltamashShaikh — APPROVED** [review]
@snake14 Left just 1 comment and its very optional, rest looks good.

File: Controller.php (hunk 2 above)
Comment (snake14) [reply Controller.php:179]: From this context, that would make sense, but I pulled the link directly from the Google Doc detailing the changes. Also, the translated string that contains the anchor is 'and watch the [training video series](https://matomo.org/tag-manager-training/).', which seems more general and mentions the series and not the specific video.

**Comment by snake14** [comment]
Thank you @AltamashShaikh . I think I'll leave the links as they are since I used what was provided in the Google Doc detailing the changes and because the general link seems to match the wording displayed with the anchor tag better.

**Comment by justinvelluppillai** [comment]
@snake14 since @AltamashShaikh's suggestions are optional and you have answered with good reason as to why you will not make the changes you can feel free to merge this now 👍🏽

**Comment by snake14** [comment]
Thank you @justinvelluppillai . I'll merge it then.

**Pull request merged**